src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -137,16 +137,14 @@ public MainTable(MainTableDataModel model,
             }
         }
         */
-       mainTablePreferences.getColumnPreferences().getColumnSortOrder().forEach(columnModel ->
+        columnPreferences.getColumnSortOrder().forEach(columnModel ->
                 this.getColumns().stream()
                     .map(column -> (MainTableColumn<?>) column)
                     .filter(column -> column.getModel().equals(columnModel))
                     .findFirst()
                     .ifPresent(column -> this.getSortOrder().add(column)));
 
-        if (mainTablePreferences.getResizeColumnsToFit()) {
-            this.setColumnResizePolicy(new SmartConstrainedResizePolicy());
-        }
+        this.setColumnResizePolicy(new SmartConstrainedResizePolicy());
```

Comment:
We removed the preference option completely - since this was a hidden feature. That should have been a visible toggle button and not hidden in the preferences.

Moreover #967 said that there should be a single behavior without the need of a configuration flat. We discussed it in length back than - and the implementors think, this is still a good thing.

Reply:
We added a guard to really cover both cases

Reply:
So you don't want to give users the option for the infinite-column behavior that you were supporting above? I don't see any harm in still having the checkbox. But since I don't want to argue again for keeping code while you want to remove it, removing is also fine with me ;-)